compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1260,7 +1260,15 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
 
          if (optimizer)
             {
-            optimizer->optimize();
+            try
+               {
+               optimizer->optimize();
+               }
+            catch (const TR::CompilationException &e)
+               {
+               bool success = comp->fe()->handleExceptionThrownFromILGen();
+               _methodFlags.set(IlGenSuccess, success);
```

Comment:
Since the _methodFlags are really for cross-compilation communication does this do the right thing for peeking? I would assume the flag would be reset by the call to do a peek, but can you confirm this is the case?